src/dataStructures/binarySearchTree.js

```diff
@@ -0,0 +1,53 @@
+const TreeNode = require('./treeNode.js');
+
+module.exports = class BinarySearchTree {
+  constructor(args) {
+    this.head = null;
+    if (Array.isArray(args) && args.length !== 0) {
+      this.addNodesFromArray(args);
+    } else if (Number.isInteger(args)) {
+      this.addNode(args);
+    } else {
+      console.log('Sorry, please check your input params.');
```

Comment:
But if I pass in `null` to the constructor, it will output with `Sorry, please check your input params.`, no?